src/plugins/gs-plugin-external-appstream.c

```diff
@@ -94,24 +105,102 @@ create_tmp_file (GsPlugin *plugin,
 	return FALSE;
 }
 
+static char *
+get_modification_date (const char *file_path)
```

Comment:
I think this would be a bit easier with GIO (you could query for the `G_FILE_ATTRIBUTE_TIME_MODIFIED` attribute) but mostly using `GDateTime`, which should avoid the `setlocale()` calls.